### PR TITLE
fix(DiscordJS): fix the way that the client filler gets the shard ID in v12

### PR DESCRIPTION
**Pull request by EndBug**
I noticed that the [v12 `ShardClientUtil`](https://discord.js.org/#/docs/main/v12/class/ShardClientUtil) does not contain the ID of the shard you're using. I figured out that the only way to get the shard id is to get a guild and take it from there. 

I'm not sure about this though: that strategy relies on the fact that every guild cached will have the same shard ID. Are we sure about that? Is it possible that selecting a random guild will result in the wrong shard ID? Is there a better alternative (like maybe taking the most used shard ID)?

Let me know what you think

### CODE REVIEW

**Comment by Snazzah** [comment]
You can read off of `client.options.shards` (shard ID) and `client.options.shardCount`.

**Comment by EndBug** [comment]
I've updated the getter, is it better now?

File: src/Interface/Clients/DiscordJS.ts [hunk 1]
```diff
@@ -36,11 +36,34 @@ export default class DiscordJS extends ClientFiller {
...
+        id: this.client.shard.ids[0],
+        count: this.client.shard.count
+      }
+
+    if (typeof this.client.options?.shards == 'number')
+      // v12 supported, manual
+      return {
+        id: this.client.options.shards,
+        count: this.client.options.shardCount
+      }
```
Comment (Snazzah) [inline src/Interface/Clients/DiscordJS.ts:65]: This is unnecessary, sorry about the confusion on my part.
```suggestion
```

**Review by Snazzah — CHANGES_REQUESTED** [review]
After this, LGTM

**Pull request merged**